src/game/HighscoreList.js

```diff
@@ -114,21 +126,55 @@ class HighscoreList {
         bg.endFill();
 
         const name = new PIXI.Text(val.name, TEXT_STYLE);
-        name.x = TEXT_PADDING;
+        name.x = TEXT_PADDING * 2 + ICON_SIZE;
         name.y = TEXT_PADDING;
 
+        // Circle icon
+        const img = this.game.basicResources.circle;
+        const icon = new PIXI.Sprite(img);
+        icon.height = ICON_SIZE;
+        icon.width = ICON_SIZE;
+        icon.x = this.rect_height / 2;
+        icon.y = this.rect_height / 2;
+
+        icon.anchor.set(0.5, 0.5);
+        const bgColor = Number.parseInt(styles[val.id].backgroundColor.substr(1), 16);
+        icon.tint = bgColor;
+
+        const iconPath = iconData[styles[val.id].iconID].img;
+        const iconName = iconData[styles[val.id].iconID].name;
+        this.game.resourceServer
+          .requestResources([
+            {
+              name: iconName,
+              path: iconPath,
+            },
+          ])
+          .then(res => {
+            const sprite = new PIXI.Sprite(res[iconName]);
+            const iconColor = Number.parseInt(styles[val.id].iconColor.substr(1), 16);
+            sprite.tint = iconColor;
+            sprite.anchor.set(0.5, 0.5);
+
+            sprite.height = SPRITE_SIZE;
+            sprite.width = SPRITE_SIZE;
+
+            icon.addChild(sprite);
+          });
+
```

Comment:
This code is essentially a duplicate of the code that creates the graphic for the PlayerCircle. Maybe reuse the methods? Essentially `new PlayerCircle(null, res).graphic` should be enough and have no unintended consequence. (Not currently the case, but that should probably be changed!)

Reply:
The idea was to avoid the overhead of creating an entire game entity. I see now that this became a bit messy though. Will fix